test/e2e/farm_build_test.go

```diff
@@ -0,0 +1,1379 @@
+			cmd := exec.Command(podmanStaticTest.PodmanBinary, "stop", regName)
+
+			session, err := Start(cmd, GinkgoWriter, GinkgoWriter)
+
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("%q failed to execute", podmanStaticTest.PodmanBinary))
+			Eventually(session, DefaultWaitTimeout).Should(Exit(0))
+		}
+		/*##########################################################################################################*/
+
+		var skopeo = func(image string) *PodmanSessionIntegration {
+
+			// skopeo inspect --tls-verify=false  --raw docker://img.pegortech.co.uk/fedora would typically produce output:
+			//
+			// {"schemaVersion":2,"mediaType":"application/vnd.oci.image.index.v1+json","manifests":[{"mediaType":"appli\
+			// cation/vnd.oci.image.manifest.v1+json","digest":"sha256:6dbd1777b7dfa0d6b595d3220cc05923eb9833d6e49263b65\
+			// 32c0ea182765017","size":612,"platform":{"architecture":"arm64","os":"linux","variant":"v8"}},{"mediaType"\
+			// :"application/vnd.oci.image.manifest.v1+json","digest":"sha256:6afa31388822c02a0452e190617affa6257d3c28ac5\
+			// 4681ab0ab35193ab86df3","size":680,"platform":{"architecture":"amd64","os":"linux"}}]}
+
+			var SkopeoBinary = "skopeo"
+
+			var dockerImage = fmt.Sprintf("docker:%s%s", `//`, image)
+			var skopeoInspect = []string{"inspect", "--tls-verify=false", "--raw", dockerImage}
+
+			var session = SystemExec(SkopeoBinary, skopeoInspect)
+
+			return &PodmanSessionIntegration{session}
+		}
+		/*##########################################################################################################*/
+
+		// the fields below are accessed by ginkgo by reflection, so need to be exported (Capitalized)
+		type Sessions struct {
+			Bld      *PodmanSessionIntegration
+			Manifest *PodmanSessionIntegration
+			Skopeo   *PodmanSessionIntegration
+		}
+		/*##########################################################################################################*/
+
+		var scanSessionForBuiltImageId = func(session *PodmanSessionIntegration, bldSpec build) (string, error) {
+			// Function attempts to extract the Id of the image built for the given bld spec
+			// from the stdout of the farm build (as captured in the session).
+			// Returns an empty string if no match is found.
+
+			var regExpEscape = func(inp string) string {
+				// User input in the form of Connection names, could potentially contain regexp
+				// metachars, which we will need to escape. There has to be a more elegant
+				// way of going this!.
+				inp = strings.ReplaceAll(inp, `\`, `\\`)
+				for _, metachar := range []byte("(){}[].?+^$|*") {
+					inp = strings.ReplaceAll(inp, string(metachar), `\`+string(metachar))
+				}
+				return inp
+			}
+
+			// Want to chop a max of two delimited items from the provided arch string
+			// i.e. linux/amd64/v8/garbage --> linux, amd64, v8/garbage (though we assume there
+			// will be no garbage!)
+
+			remainder := bldSpec.arch
+			opSys, remainder, _ := strings.Cut(remainder, "/")
```

Comment:
opSys should be names os I guess? kinda hard to figure out what opSys means otherwise